react-components/src/data/search/QueryBuilder.js

```diff
@@ -0,0 +1,106 @@
+import build from "../../util/DebugIDUtil";
+import Condition from "./queryBuilder/Condition";
+import ids from "./ids";
+import messages from './messages';
+import SaveSearchButton from "./SaveSearchButton";
+import styles from "./styles";
+import withI18N, { getMessage } from "../../util/I18NWrapper";
+import withStoreProvider from "../../util/StoreProvider";
+
+import Button from '@material-ui/core/Button';
+import PropTypes from "prop-types";
+import React, { Component } from 'react';
+import { reduxForm, Field } from "redux-form";
+import { withStyles } from "@material-ui/core/styles";
+
+/**
+ * A form which allows users to build their own custom data search queries
+ */
+class QueryBuilder extends Component {
+    constructor(props) {
+        super(props);
+
+        this.handleSubmitForm = this.handleSubmitForm.bind(this);
+        this.handleSaveSearch = this.handleSaveSearch.bind(this);
+    }
+
+    handleSubmitForm(values) {
+        this.props.presenter.onSearchBtnClicked(values);
+    }
+
+    handleSaveSearch(values) {
+        let initialValues = this.props.initialValues;
+        let originalName = initialValues ? initialValues.label : null;
+        this.props.presenter.onSaveSearch(values, originalName);
+    }
+
+    render() {
+        let {
```

Comment:
Just being picky here but this declaration should be `const` instead of `let` because props are readonly. Mentioning only once here but applies all `let` declarations with props.